src/server/server.c

```diff
@@ -0,0 +1,53 @@
+#include "server.h"
+
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "../util/util.h"
+
+server_t* init_server(struct sockaddr_in ip_addr, int max_backlog) {
```

Comment:
I think we need a `free_server()` somewhere to call it if we `malloc` for the server struct